utilities_common/multi_asic.py

```diff
@@ -0,0 +1,159 @@
+        self.current_namespace = None
+
+    def connect_dbs_for_ns(self,namespace=DEFAULT_NAMESPACE):
+        '''
+        The function connects to the DBs for a given namespace and
+        returns the handle
+        If no namespace is provide, it will connect to the db in the
+        default namespace.
+        In case of multi ASIC, the default namespace in the database instance running the on the host
+        In case of single ASIC, the namespace has to DEFAULT_NAMESPACE
+        '''
+
+        db = swsssdk.SonicV2Connector(use_unix_socket_path=True, namespace=namespace)
+        db.connect(db.APPL_DB)
+        db.connect(db.CONFIG_DB)
+        db.connect(db.STATE_DB)
+        db.connect(db.COUNTERS_DB)
+        db.connect(db.ASIC_DB)
+        return db
+
+    def connect_config_db_for_ns(self, namespace=DEFAULT_NAMESPACE):
+        '''
+        The function connects to the config DB for a given namespace and
+        returns the handle
+        If no namespace is provide, it will connect to the db in the
+        default namespace.
+        In case of multi ASIC, the default namespace in the database instance running the on the host
+        In case of single ASIC, the namespace has to DEFAULT_NAMESPACE
+        '''
+        config_db = ConfigDBConnector(use_unix_socket_path=True, namespace=namespace)
+        config_db.connect()
+        return config_db
+
+    def is_object_internal(self, object_type, cli_object):
+        '''
+        The function check if a CLI object is internal and returns true or false.
+        For single asic, this function is not applicable
+        '''
+        if object_type == PORT_OBJ:
+            return is_port_internal(cli_object)
+        elif object_type == PORT_CHANNEL_OBJ:
+            return is_port_channel_internal(cli_object)
+        elif object_type == BGP_NEIGH_OBJ:
+            return is_bgp_session_internal(cli_object)
+
+    def skip_display(self, object_type, cli_object):
+        '''
+        The function determines if the passed cli_object has to be displayed or not.
+        returns true if the display_option is external and  the cli object is internal.
+        returns false, if the cli option is all or if it the platform is single ASIC.
+
+        '''
+        if not is_multi_npu():
+            return False
+        if self.display_option == DISPLAY_ALL:
+            return False
+        return self.is_object_internal(object_type, cli_object)
+
+    def get_ns_list_based_on_options(self):
+        ns_list = []
+        if not is_multi_npu():
+            return [DEFAULT_NAMESPACE]
+        else:
+            namespaces = get_all_namespaces()
+            if self.namespace_option is None:
+                if self.display_option ==  DISPLAY_ALL:
+                    ns_list = namespaces['front_ns'] + namespaces['back_ns']
+                else:
+                    ns_list = namespaces['front_ns']
+            else:
+                ns_list = [self.namespace_option]
```

Comment:
We should validate namespace here. With this class we don't expect CLIs to get namespace list and validate the namespace.

Reply:
added validation in the latest commit